third_party/nvfuser/csrc/lower2device.cpp

```diff
@@ -132,6 +132,46 @@ class KIRCleaner : public OptOutDispatch {
   bool is_nop_ = false;
 };
 
+// Convert bar sync to __syncthreads()
+class ConvertAlignedBlockSync : kir::IrVisitor {
+ public:
+  static std::vector<Expr*> run(std::vector<Expr*> exprs) {
+    ConvertAlignedBlockSync converter;
+    converter.handle(exprs);
+    return exprs;
+  }
+
+ private:
+  using kir::IrVisitor::handle;
+
+  void handle(kir::BlockSync* sync) final {
+    // Inspect all the scope expressions
+    for (auto expr : scope_exprs_) {
+      // All predicates we have today are can be thread dependent so avoid this
+      // in general.
+      if (expr->isA<kir::IfThenElse>()) {
```

Comment:
Why can't you just use `dependsonthreadNamedScalars` for this case as well?

Reply:
Good point, changed